dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtension.cs

```diff
@@ -0,0 +1,127 @@
+    public static KernelFunction CreateFunctionFromPrompty(
+        this Kernel _,
+        string promptyPath,
+        IPromptTemplateFactory? promptTemplateFactory = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        var text = File.ReadAllText(promptyPath);
+
+        promptTemplateFactory ??= new LiquidPromptTemplateFactory(); // use liquid template factory by default
+
+        // create PromptTemplateConfig from text
+        // step 1
+        // retrieve the header, which is in yaml format and put between ---
+        //
+        // e.g
+        // file: chat.prompty
+        // ---
+        // name: Contoso Chat Prompt
+        // description: A retail assistant for Contoso Outdoors products retailer.
+        // authors:
+        //   - XXXX
+        // model:
+        //   api: chat
+        //   configuration:
+        //     type: azure_openai
+        //     azure_deployment: gpt - 35 - turbo
+        //     api_version: 2023 - 07 - 01 - preview
+        //   parameters:
+        //     tools_choice: auto
+        //     tools:
+        //     -type: function
+        //       function:
+        //         name: test
+        //         description: test function
+        //         parameters:
+        //           properties:
+        //             location:
+        //               description: The city and state or city and country, e.g.San Francisco, CA
+        //                 or Tokyo, Japan
+        // ---
+        // ... (rest of the prompty content)
+
+        var splits = text.Split(["---"], StringSplitOptions.RemoveEmptyEntries);
+        var yaml = splits[0];
+        var content = splits[1];
+
+        var deserializer = new DeserializerBuilder().Build();
+        var prompty = deserializer.Deserialize<PromptyYaml>(yaml);
+
+        // step 2
+        // create a prompt template config from the prompty object
+        var promptTemplateConfig = new PromptTemplateConfig
+        {
+            Name = prompty.Name, // TODO: sanitize name
+            Description = prompty.Description,
+            Template = content,
+        };
+
+        PromptExecutionSettings defaultExecutionSetting = prompty.Model?.ModelConfiguration?.ModelType switch
+        {
+            ModelType.azure_openai or ModelType.openai => new OpenAIPromptExecutionSettings()
+            {
+                ResponseFormat = prompty.Model?.Parameters?.ResponseFormat == "json_object" ? ChatCompletionsResponseFormat.JsonObject : null,
+                Temperature = prompty.Model?.Parameters?.Temperature ?? 1.0,
+                TopP = prompty.Model?.Parameters?.TopP ?? 1.0,
+                MaxTokens = prompty.Model?.Parameters?.MaxTokens,
+                Seed = prompty.Model?.Parameters?.Seed,
+                ModelId = prompty.Model?.ModelConfiguration?.AzureDeployment,
+            },
+            _ => throw new NotSupportedException($"Model type '{prompty.Model?.ModelConfiguration?.ModelType}' is not supported."),
```

Comment:
It seems like unfortunate layering that the prompty library would need to have a dependency on every connector for every supported service. Seems like it'd be better to just populate the base PromptExecutionSettings and then allow the provider to extract from it when it needs. I believe the connectors already do this, e.g.
https://github.com/microsoft/semantic-kernel/blob/e0be61601d5e6475d45ff79399bd4b3d22135e80/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L129
https://github.com/microsoft/semantic-kernel/blob/e0be61601d5e6475d45ff79399bd4b3d22135e80/dotnet/src/Connectors/Connectors.OpenAI/OpenAIPromptExecutionSettings.cs#L328-L334
@markwallace-microsoft, do you agree?

Reply:
Presumably whatever we do here for service-agnostic support for specifying tool calling should also come into play here, as we'd want to channel the tool call information through the base PromptExecutionSettings into the connector. cc: @SergeyMenshykh

Reply:
Completely agree that the base PromptExecutingSettings should be used so this will work with all connectors.

Reply:
Resolved by passing openai setting through `ExtensionData`